activestorage/test/models/attached/many_test.rb

```diff
@@ -866,6 +866,36 @@ def self.name; superclass.name; end
     assert_equal 67, image.height
   end
 
+  test "creating variation with dynamic variation method" do
+    john = User.create!(name: "John")
+
+    assert_no_enqueued_jobs only: ActiveStorage::TransformJob do
+      @user.highlights_with_dynamic.attach fixture_file_upload("racecar.jpg")
+      john.highlights_with_dynamic.attach fixture_file_upload("racecar.jpg")
+    end
+
+    josh_variantion = @user.highlights_with_dynamic.first.variant(:thumb).variation
+    john_variantion = john.highlights_with_dynamic.first.variant(:thumb).variation
+
+    assert_equal({ format: "jpg", resize_to_limit: [100, 100] }, josh_variantion.transformations)
+    assert_equal({ format: "jpg", resize_to_limit: [200, 200] }, john_variantion.transformations)
+  end
+
+  test "creating variation with dynamic variation proc" do
+    john = User.create!(name: "John")
+
+    assert_no_enqueued_jobs only: ActiveStorage::TransformJob do
+      @user.highlights_with_dynamic.attach fixture_file_upload("racecar.jpg")
+      john.highlights_with_dynamic.attach fixture_file_upload("racecar.jpg")
+    end
+
+    josh_variantion = @user.highlights_with_dynamic.first.variant(:thumb_proc).variation
+    john_variantion = john.highlights_with_dynamic.first.variant(:thumb_proc).variation
+
+    assert_equal({ format: "jpg", resize_to_limit: [100, 100] }, josh_variantion.transformations)
+    assert_equal({ format: "jpg", resize_to_limit: [200, 200] }, john_variantion.transformations)
+  end
```

Comment:
typo fix: s/variantion/variation

```suggestion
    josh_variation = @user.highlights_with_dynamic.first.variant(:thumb).variation
    john_variation = john.highlights_with_dynamic.first.variant(:thumb).variation

    assert_equal({ format: "jpg", resize_to_limit: [100, 100] }, josh_variation.transformations)
    assert_equal({ format: "jpg", resize_to_limit: [200, 200] }, john_variation.transformations)
  end

  test "creating variation with dynamic variation proc" do
    john = User.create!(name: "John")

    assert_no_enqueued_jobs only: ActiveStorage::TransformJob do
      @user.highlights_with_dynamic.attach fixture_file_upload("racecar.jpg")
      john.highlights_with_dynamic.attach fixture_file_upload("racecar.jpg")
    end

    josh_variation = @user.highlights_with_dynamic.first.variant(:thumb_proc).variation
    john_variation = john.highlights_with_dynamic.first.variant(:thumb_proc).variation

    assert_equal({ format: "jpg", resize_to_limit: [100, 100] }, josh_variation.transformations)
    assert_equal({ format: "jpg", resize_to_limit: [200, 200] }, john_variation.transformations)
  end
```